cfme/tests/infrastructure/test_host.py

```diff
@@ -430,3 +432,33 @@ def test_infrastructure_hosts_navigation_after_download(
     hosts_view.navigation.select("Compute")
     provider_view = provider.create_view(InfraProviderDetailsView)
     assert provider_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1634794, forced_streams=["5.10"])], automates=[1634794])
+def test_infrastructure_hosts_no_edit_no_warning(appliance, setup_provider_min_hosts, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/6h
+    Bugzilla:
+            1634794
+    """
+    refresh = 2
+    my_slice = slice(0, refresh, None)
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    num_hosts = hosts_view.entities.paginator.items_amount
+    if num_hosts < refresh:
+        pytest.skip('not enough hosts in appliance UI to run test')
```

Comment:
This block keeps getting repeated, consider a module scoped global fixture (cfme/fixtures/) that skips in setup for any test functions requiring multiple hosts. Then we're not checking at the start of every function, or duplicating code.

Reply:
Did you see the fixture setup_provider_min_hosts.  I really thought we worked this out weeks ago. I check in the yaml files for the number of possible hosts, then I check again that these hosts actually appear in the GUI.

Reply:
@prichard77 yep, @mshriver is saying that the checking of the UI should be done in a fixture too, to avoid code duplication.

Reply:
I moved this code up into the setup_provider_min_hosts fixture. It's only used by the tests that need it.

Reply:
I also had to update test_infrastructure_hosts_refresh_multi to use the updated fixture. There are also tests in other Prs that will use the updated fixture.